guides/plugins/plugins/framework/event/finding-events.md

```diff
@@ -0,0 +1,184 @@
+$this->eventDispatcher->dispatch($someEvent, $someEvent->getName());
+```
+
+This is an event that's being fired manually, which you can react upon.
+Make sure to always have a look at the event class itself in order to find out which information it contains.
+
+When subscribing to those events, your event listener method will have access to the previously created event instance. 
+
+```php
+public static function getSubscribedEvents(): array
+{
+    return [
+        SomeEvent::class => 'registeringToSomeEvent'
+    ];
+}
+
+public function registeringToSomeEvent(SomeEvent $event): void
+{
+}
+```
+
+The [next section](#Specifically searching for events) will cover how to find those events without randomly stumbling upon them.
+
+### Specifically searching for events
+
+If you're really looking for a fitting event for your purpose, you might want to directly search for them.
+This can be done by searching through the `<shopware root>/platform/src` or the `<shopware root>/vendor/shopware/platform/src` directory,
+depending on whether you're using the [development](https://github.com/shopware/development) or the [production template](https://github.com/shopware/production).
+Use one of the following search terms:
+
+- `extends NestedEvent`: This way you will find the events themselves.
+- `extends Event`: This way you will find the events themselves.
+- `->dispatch`: Here you will find all the occurrences where the events are actually being fired.
+
+### Looking at the service definition
+
+Every service, that wants to fire an event sooner or later, needs access to the `event_dispatcher` in order to do so.
+
+Hence, you can have a look at all the service definitions for the [dependency injection container](../../plugin-fundamentals/dependency-injection.md)
+and therefore quickly figure out, which services and classes are having access to the said `event_dispatcher`:
+
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="Some\Service">
+            <argument type="service" id="Another/Service"/>
+            <argument type="service" id="event_dispatcher"/>
+        </service>
+    </services>
+</container>
+```
+
+Therefore, you could simply search for occurrences of the `event_dispatcher` in the respective `.xml` files.
+
+You can also do this the other way around, by having a look at the service's constructor parameters.
+
+```php
+public function __construct(
+    Some\Service $someService,
+    EventDispatcherInterface $eventDispatcher
+) {
+    $this->someService = $someService;
+    $this->eventDispatcher = $eventDispatcher;
+}
+```
+
+If it's having access to the `EventDispatcherInterface`, you're most likely going to find at least one event being fired
+in that service.
```

Comment:
There is an other way that I really like. The symfony web profiler. It will capture all dispatched events. They have a nice Web UI for that and makes it easy to see what kind of events happen when you click a certain button on the web interface or API call. It is just clicking the UI and see what got tracked. That is a quite less safari way.

Reply:
Will definitely include the profiler! Thanks!

Reply:
Good explanation. The picture is quite helpful